includes/sanitizers/class-amp-video-sanitizer.php

```diff
@@ -179,16 +184,16 @@ private function filter_attributes( $attributes ) {
 
 		foreach ( $attributes as $name => $value ) {
 			switch ( $name ) {
+				case 'poster':
```

Comment:
This is an edge case. But do you think the `poster` should be forced to HTTPS like it is here?

It looks like HTTP is allowed for `poster`. For example, passing [this amp-video](https://ampb.in/#-LHnv3JlnbLeBupB44Co) with an HTTP `poster` to the [AMP Validator](https://validator.ampproject.org/) results in a valid page.

The HTTPS request might fail if we upgrade it from HTTP. But if we're already upgrading the `src` to HTTPS, and that might be on the same domain. So this probably wouldn't be a common issue.

Reply:
Good observation, @kienstra.  You're right, it does validate.  Therefore, we do not need to force the `poster` to HTTPS per the edge case you identified.  I'll roll that part back.